src/cargo/ops/cargo_add/mod.rs

```diff
@@ -758,6 +761,13 @@ fn populate_dependency(mut dependency: Dependency, arg: &DepOp) -> Dependency {
             dependency.optional = None;
         }
     }
+    if let Some(value) = arg.public {
+        if value {
+            dependency.public = Some(true);
+        } else {
+            dependency.public = None;
+        }
+    }
```

Comment:
As currently implemented, when you inherit a public dependency, you can make it non-public.  I'm double checking those semantics at https://github.com/rust-lang/rust/issues/44663#issuecomment-1826474620

If we keep those semantics, we'll want to allow setting `dependency.public = Some(false);` if the dependency is inherited and it is public.  If we support that, we should have a test case for it.

Reply:
Let's go ahead and assume we won't be inheriting `public` but that assumption should be noted in the PR description and tracking issue to make sure we revisit this when a decision is made.